deps/fdasrsf/dp_nbhd.h

```diff
@@ -0,0 +1,77 @@
+#ifndef DP_NBHD_H
+#define DP_NBHD_H 1
+
+#ifndef DP_NBHD_DIM
+#define DP_NBHD_DIM 7 
+#endif
+
+#if DP_NBHD_DIM == 17
+int dp_nbhd[][2] = {
```

Comment:
OMG, what the hell is this??

Reply:
I didn't change the original code following the next rule 😆 

![image](https://user-images.githubusercontent.com/16774925/54088396-0f0ce180-435e-11e9-9dbd-a867021e9440.png)

In `dp_grid.c` a grid it is explored evaluating an energy function, this file contains the coordinates of the grid flattened. I guess they did it for performance reasons.

By the moment we only use one algorithm, but the original code contains more modules which uses this grid (maybe in a future they could be implemented), I guess this is the reason why the grid coordinates are in a global variable in a header.

Reply:
But why the arbitrary limits? Why not providing an inline function to compute it instead? The performance should be almost the same (or even better, as you are playing nicely with the cache). This code looks COMPLETELY awful.

Reply:
If we have proper tests, we should be able to change it safely.

Reply:
Tests were added in de19847. 
Looking at the coordinate lists in detail I realised that they are not simply the Cartesian product, for optimisation reasons it is not necessary to evaluate all the points, which complicates the dynamic generation of the list with an inline function.

These are the coordinates in the list (colours indicate different sizes):
![Figure_6](https://user-images.githubusercontent.com/16774925/54493958-ae3d5600-48d5-11e9-96bf-91c135cff669.png)

Maybe the list may be initialised at the beginning of the algorithm or we can use always the same size.